### ISSUE TITLE
Identifying web apps

### ISSUE THREAD

**Issue by dgstpierre** [WICG/ua-client-hints#280]
In the current user agent we have an indicator of 'wv' that assists with identifying web apps.  As well as the web app adding it's identity.  I don't see anything that accounts for this, looks like lost functionality.

**Comment by miketaylr** [comment]
> In the current user agent we have an indicator of 'wv' that assists with identifying web apps.

Can you clarify what you mean by web apps? Do you mean native apps with a webview component?

**Comment by dgstpierre** [comment]
Yes that is what I mean.  The user agent currently has 'wv' in it to indicate.

**Comment by ronancremin** [comment]
Note that the Chromium blog [stated](https://blog.chromium.org/2021/05/update-on-user-agent-string-reduction.html) that "have no plans to change the User-Agent string on Android WebView or Chrome for iOS at this time, but will make public updates if and when that changes."

So, by default, apps using the default Chrome-based Android webview ("Android System WebView") will presumably continue to behave as before—perhaps @miketaylr can confirm.

**Comment by miketaylr** [comment]
Correct, this is not an announcement that anything is changing from what is previously announced. To be clear, I'm saying it may be useful to consider a specific type of client hint to a webview.

**Comment by ronancremin** [comment]
Understood, thanks.

**Comment by dgstpierre** [comment]
I wanted to check in on this with the user agent reduction coming for mobile devices in a few months.  Will the user agent reduction remove wv?  Or will it still be there to fall back on if user agent hints doesn't add a way to support it?  Thanks

**Comment by Sora2455** [comment]
Why couldn't the browser just be listed as "Chrome WV" instead of "Chrome"?

**Comment by miketaylr** [comment]
> I wanted to check in on this with the user agent reduction coming for mobile devices in a few months. Will the user agent reduction remove wv? Or will it still be there to fall back on if user agent hints doesn't add a way to support it? Thanks

The upcoming changes won't affect webview. Right now webview doesn't support client hints, which I consider a blocker for any UA reduction effort. But that will soon change (see https://groups.google.com/a/chromium.org/g/blink-dev/c/N_Y40YIHMtI). If and when we have a plan for webview reduction, we'll be sure to make it public and give devs plenty of time to prep for it/provide feedback, etc.

**Comment by miketaylr** [comment]
See also https://github.com/WICG/ua-client-hints/issues/219#issuecomment-1749217579

> In particular we would like to see User-Agent Client Hints to indicate:
> 
> If the traffic is coming from a WebView
> The source application that the browser session is coming from

The first would be covered by a hint that expressed "hey i'm a webview", and the second would be basically X-Requested-With, right? That was recently removed AFAIU: https://chromestatus.com/feature/5160086884843520

**Comment by jross1012** [comment]
That's right, it would be great to see this supported with Client Hints.

**Comment by miketaylr** [comment]
Note to self, this seems relevant: https://android-developers.googleblog.com/2023/02/improving-user-privacy-by-requiring-opt-in-to-send-x-requested-wih-header-from-webview.html

**Comment by Tanych** [comment]
the `sec-ch-ua` client hints already support to identify the traffic coming from WebView

for android it would be like: 
* Sec-CH-UA: "Google Chrome";v="117", "Not;A=Brand";v="8", "Chromium";v="117"
for webview  it would be like:
* Sec-CH-UA: "Android WebView";v="117", "Chromium";v="117", "Not;A=Brand";v="8"

**Comment by Shivaansh-Agarwal** [comment]
The `sec-ch-ua` is available but it doesn't seem to be a reliable approach as some browsers still don't support it.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-UA#browser_compatibility

**Comment by bvattikonda** [comment]
Any update on this issue? `Sec-CH-UA` supports identifying traffic that is coming from webview. But, the issue of using client hints to identify source application remains unresolved.

**Comment by Tanych** [comment]
the app can override the user agent metadata to give their own client hints, e.g. they can add their own brand version: https://developer.android.com/reference/androidx/webkit/WebSettingsCompat#setUserAgentMetadata(android.webkit.WebSettings,androidx.webkit.UserAgentMetadata)

**Comment by bvattikonda** [comment]
Our request here is to make the contents of what used to be `X-Requested-With` a part of high entropy client hints by default. Unlike `X-Requested-With` the header will not be sent by default because it would be a high entropy client hint but at the same time it would be available for websites that would benefit from the value of the header.
